### PR TITLE
Display daily data

### PR DESCRIPTION
**Pull request by shadrackkibet**
In progress!

### CODE REVIEW

**Comment by shadrackkibet** [comment]
@dannyparsons  i have tried adding stations to  factors but this seems to be taking time to produce the table. Do we really need the station receiver  in the table part of this dialog now?

**Comment by dannyparsons** [comment]
Yes we do need the station receiver because we have to look at each station separately. However, we also need to think of ways to do this faster. I suggest for testing you use a small subset of years so it's not too slow.

**Comment by shadrackkibet** [comment]
@africanmathsinitiative/developers this is ready for testing. 

To work with stations use a small subset  of years in your dataset.

**Comment by shadrackkibet** [comment]
@maxwellfundi  please test this.

**Comment by shadrackkibet** [comment]
@dannyparsons  this is ready for you to review.

File: instat/dlgDisplayDailyData.vb [hunk 1]
```diff
@@ -204,11 +194,15 @@ Public Class dlgDisplayDailyData
...
         TestOkEnabled()
     End Sub
 
-    Private Sub ucrReceiverFactorby_ControlValueChanged(ucrChangedControl As ucrCore) Handles ucrReceiverFactorby.ControlValueChanged
-        clsSummaryTableFunction.AddParameter("factors", ucrReceiverFactorby.GetVariableNames)
+    Private Sub ucrReceiverYear_ControlValueChanged(ucrChangedControl As ucrCore) Handles ucrReceiverYear.ControlValueChanged, ucrReceiverDay.ControlValueChanged, ucrReceiverMonth.ControlValueChanged, ucrReceiverStations.ControlValueChanged
+        If Not ucrReceiverStations.IsEmpty Then
+            clsSummaryTableFunction.AddParameter("factors", "c(" & ucrReceiverStations.GetVariableNames & "," & ucrReceiverYear.GetVariableNames & "," & ucrReceiverMonth.GetVariableNames & "," & ucrReceiverDay.GetVariableNames & ")")
+        Else
+            clsSummaryTableFunction.AddParameter("factors", "c(" & ucrReceiverYear.GetVariableNames & "," & ucrReceiverMonth.GetVariableNames & "," & ucrReceiverDay.GetVariableNames & ")")
```
Comment (dannyparsons) [inline instat/dlgDisplayDailyData.vb]: Can you instead do this automatically with c() as a function and the receivers having additional codes for this function?

**Comment by shadrackkibet** [comment]
This is the previous comment From Danny at #3761 

This is the R code for the tables:

`InstatDataObject$summary_table(data_name="Dodoma", summaries=sum_label, columns_to_summarise = c("Rain"), factors=c("Year", "Month", "Day"), n_column_factors=1, include_margins=TRUE, rnames=FALSE, as_html = FALSE, page_by = "Year", store_results = FALSE)`

Some of these parameter will NOT be options on the dialog. summaries is always` sum_label`, `page_by` is always the year column and `n_column_factors = 1 `always. `stored_results = FALSE` always as we are not doing any "new" summaries.

The order the factors appear in the list is important so that the rows are days and columns are months in the table.
`include_margins`, `rnames` and `as_html` are all options. Check with @rdstern about the defaults for these.

@rdstern what are the defaults for this parameters `include_margins`, `rnames` and `as_html`?

**Comment by shadrackkibet** [comment]
**This is the current designer layout!**

image
image

File: instat/dlgDisplayDailyData.vb (hunk 1 above)
Comment (shadrackkibet) [reply instat/dlgDisplayDailyData.vb]: This is now sorted out thanks!

**Comment by rdstern** [comment]
I like the look of the current design of the daily data dialogue.  I really look forward to testing it.

**Comment by dannyparsons** [comment]
let's `include_margins = TRUE`, `rnames = FALSE` and `as_html = TRUE` as default to start with
@rdstern may then want to change this after using it

**Comment by shadrackkibet** [comment]
@africanmathsinitiative/developers this is ready for re-review.

**Comment by shadrackkibet** [comment]
@dannyparsons this is ready for you to review.

File: instat/dlgDisplayDailyData.vb [hunk 2]
```diff
@@ -130,24 +130,35 @@ Public Class dlgDisplayDailyData
...
 
+        clsConcFunction.SetRCommand("c")
         clsDisplayDailyGraphFunction.AddParameter("rug_colour", Chr(34) & "red" & Chr(34))
         clsDisplayDailyGraphFunction.AddParameter("bar_colour", Chr(34) & "blue" & Chr(34))
-        clsSummaryTableFunction.AddParameter("summaries", "sum_label")
-        clsSummaryTableFunction.AddParameter("n_column_factors", "1")
-        clsSummaryTableFunction.AddParameter("stored_results", Chr(34) & "FALSE" & Chr(34))
+        clsSummaryTableFunction.AddParameter("summaries", "sum_label", iPosition:=1)
+        clsSummaryTableFunction.AddParameter("n_column_factors", "1", iPosition:=4)
+        clsSummaryTableFunction.AddParameter(" include_margins", "TRUE", iPosition:=5)
```
Comment (dannyparsons) [inline instat/dlgDisplayDailyData.vb]: You've got a space you don't need here before `include`.

File: instat/dlgDisplayDailyData.vb [hunk 3]
```diff
@@ -130,24 +130,35 @@ Public Class dlgDisplayDailyData
...
+        clsSummaryTableFunction.AddParameter("rnames", "FALSE", iPosition:=6)
+        clsSummaryTableFunction.AddParameter("as_html ", "TRUE", iPosition:=7)
```
Comment (dannyparsons) [inline instat/dlgDisplayDailyData.vb]: and an extra space at the end of html here.

File: instat/dlgDisplayDailyData.vb (hunk 2 above)
Comment (shadrackkibet) [reply instat/dlgDisplayDailyData.vb]: space removed.

File: instat/dlgDisplayDailyData.vb (hunk 3 above)
Comment (shadrackkibet) [reply instat/dlgDisplayDailyData.vb]: space removed

**Pull request merged**